src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -1551,65 +1578,64 @@ private fun checkParamAttrs(holder: RsAnnotationHolder, o: RsOuterAttributeOwner
     diagnostic.addToHolder(holder)
 }
 
-private fun PsiElement.nameOrImportedName(): String? =
-    when (this) {
-        is RsNamedElement -> name
-        is RsUseSpeck -> nameInScope
-        else -> null
-    }
-
 private val RsNamedElement.namespacesForDuplicatesCheck: Set<Namespace>
```

Comment:
Could you please add a comment for this method explaining why `namespacesForDuplicatesCheck` is different from the more general `namespaces` implementation? This method existed before this PR, but I am reading the code for the first time and for me, it seems difficult to guess

Reply:
Thanks, now it looks more clear!